papiea-sdk/typescript/__tests__/intentful_workflow_tests/intentful_workflow.test.ts

```
@@ -718,6 +718,138 @@ describe("Intentful Workflow tests", () => {
+        const sdk = ProviderSdk.create_provider(papieaUrl, adminKey, server_config.host, server_config.port);
+        try {
+            first_provider_prefix = "location_provider_intentful_2_concurrent_cas"
+            const location = sdk.new_kind(locationDataDescription);
+            sdk.version(provider_version);
+            sdk.prefix(first_provider_prefix);
+            location.on("x", async (ctx, entity, input) => {
+                await providerApiAdmin.patch(`/${sdk.provider.prefix}/${sdk.provider.version}/update_status`, {
+                    context: "some context",
+                    entity_ref: {
+                        uuid: metadata.uuid,
+                        kind: kind_name
+                    },
+                    status: { x: entity.spec.x }
+                })
+            })
+            location.on_create(async (ctx, entity) => {
+                const { metadata, spec } = entity
+                await ctx.update_status(metadata!, spec!)
+            })
+            await sdk.register();
+            const kind_name = sdk.provider.kinds[0].name;
+            const { data: { metadata, spec } } = await entityApi.post(`${ sdk.entity_url }/${ sdk.provider.prefix }/${ sdk.provider.version }/${ kind_name }`, {
+                spec: {
+                    x: 10,
+                    y: 11
+                }
+            })
+            first_provider_to_delete_entites.push(metadata)
+            await timeout(5000)
+            const promises = []
+            for (let i = 0; i < 2; i++) {
+                promises.push(entityApi.put(`/${ sdk.provider.prefix }/${ sdk.provider.version }/${ kind_name }/${ metadata.uuid }`, {
+                    spec: {
+                        x: 20,
+                        y: 11
+                    },
+                    metadata: {
+                        spec_version: 1
+                    }
+                }))
+            }
+            const results = await Promise.all(promises)
```

This doesn't seem right to me—at least one update should have failed. I would expect this to throw a `ConflictingEntityError`.

---

Please see the comments on the PR on the expected behavior I implemented. TLDR; All the differ entities return a watcher (even if a Failed one) and the error if present is in the watcher field

---

Yeah sorry, I should have written a clearer comment—I get that's the intention here, I just don't think it's the right thing to do.

1. There is more than one way for clients to observe entity status changes—they can use the diff watcher that's returned, or they can simply poll an entity with get(). If they do the latter (which is what bmaas always does, for example), there is no indication the update failed.
2. More generally, as with any other return-value based error reporting, clients can easily ignore the return value and have no idea an error occurred. This might be idiomatic in other languages but it is not idiomatic in JavaScript and is likely to be surprising to users of the SDK, especially when other types of errors (e.g. Papiea server isn't reachable, etc.) result in exceptions.
3. To make matters worse, only _some_ entity types report conflicts through diff-watchers. Others actually will throw `ConflictingEntityError`. We should pick one behavior and stick with it rather than requiring users to understand the type of entity they're dealing with. Users should not need to know or care whether they are dealing with a differ entity or some other type of entity.

---

I think thats a good point.. Let's throw a conflict error when unable to CAS on all entity types.